lib/ansible/plugins/connection/paramiko_ssh.py

```diff
@@ -283,6 +283,7 @@ def _parse_proxy_command(self, port=22):
             for find, replace in replacers.items():
                 proxy_command = proxy_command.replace(find, str(replace))
             try:
+                proxy_command = os.path.expandvars(proxy_command)
```

Comment:
this expands variables as per the controller, not the proxy itself, if this is the case it is prefereable to just use a `lookup('env' ..` and allow configuration to be explicit as we don't want environment variable expansion in general in the options.